arch/arm/core/aarch64/swap.c

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2019 Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <kernel_internal.h>
+
+extern const int _k_neg_eagain;
+
+int arch_swap(unsigned int key)
```

Comment:
If you eventually want to enable SMP you should implement arch_switch() instead.

Reply:
I do not plan to support SMP at the moment, so implementing `arch_switch()` is something for another story.